File: templates/snake/cpp/snake.cpp
// Snake skeleton (C++). Fill in the TODOs; keep methods small and flat.
#include <utility>
#include <vector>

constexpr int WIDTH = 20;
constexpr int HEIGHT = 20;

struct Game {
    std::vector<std::pair<int, int>> snake{{0, 0}};
    std::pair<int, int> direction{0, 1};

    // Returns false when the game is over.
    bool step() {
        // TODO: read input, move snake, handle food/growth, detect collisions.
        return false;
    }
};

int main() {
    Game game;
    bool running = true;
    while (running) {
        running = game.step();
    }
    return 0;
}

File: templates/snake/python/snake.py
"""Snake skeleton (Python). Fill in the TODOs; keep functions small and flat."""


def step(snake, direction, food):
    # TODO: move the snake, handle food/growth, and return False on game over.
    return False


def main():
    snake = [(0, 0)]
    direction = (0, 1)
    food = (5, 5)
    running = True
    while running:
        running = step(snake, direction, food)


if __name__ == "__main__":
    main()
